app/Console/Commands/SyncBDclub.php

```diff
@@ -0,0 +1,220 @@
+            $ical = new ICal();
+            $ical->initString($event->getData());
+            $icalEvents = $ical->events();
+            /* @var $icevt Event */
+            foreach ($icalEvents as $icevt) {
+                $this->info($icevt->summary);
+                //$this->info($icevt->comment);
+                $extraData = json_decode($icevt->comment);
+                $this->info("extradata:".$icevt->comment);
+                //$this->info(var_dump($extraData));
+                $this->info($icevt->uid);
+                
+                /* @var $clubEvent ClubEvent */
+                $clubEvent = ClubEvent::where('external_id', '=', $icevt->uid)->first();
+                if (is_null($clubEvent)) {
+                    $this->info('Create new event for '.$icevt->summary);
+                    $clubEvent = new ClubEvent();
+                } else {
+                    $this->info('update event '.$icevt->summary);
+                }
+                
+                $clubEvent->evnt_title = $icevt->summary;
+                $clubEvent->external_id = $icevt->uid;
+                $clubEvent->evnt_is_private = !(!is_null($extraData->isactive) && $extraData->isactive == 'on');
+                $clubEvent->evnt_date_start = (new \DateTime($icevt->dtstart))->format('Y-m-d');
+                $clubEvent->evnt_date_end = (new \DateTime($icevt->dtend))->format('Y-m-d');
+                $clubEvent->evnt_time_start = '21:00:00';
+                $clubEvent->evnt_time_end = '01:00:00';
+                $clubEvent->plc_id = $section->id;
+                $clubEvent->price_normal = $extraData->price_ak_n;
+                $clubEvent->price_external = $extraData->price_ak_v;
+                $clubEvent->price_tickets_external = $extraData->price_vk_v;
+                $clubEvent->price_tickets_normal = $extraData->price_vk_n;
+                $clubEvent->save();
+                $clubEvent->showToSection()->sync($section->id);
+                
+                $clubEvent->facebook_done = $extraData->facedone!=null && $extraData->facedone == "on";
+                $clubEvent->event_url = $extraData->face;
+                $flyerDone = "Flyer erledigt: " . ($extraData->flyerdone != null ? "Ja":"Nein");
+                
+                $clubEvent->evnt_private_details = $flyerDone . "\n\n" . $extraData->notes;
+                
+                /* @var $schedule Schedule */
+                $schedule = $clubEvent->schedule;
+                if (is_null($schedule)) {
+                    $schedule = new Schedule();
+                }
+                $schedule->evnt_id = $clubEvent->id;
+                $schedule->schdl_title = $clubEvent->evnt_title;
+                $schedule->schdl_time_preparation_start = '20:00:00';
+                
+                $schedule->save();
+                $shifts = $schedule->shifts;
+                if ($shifts->isEmpty()) {
+                    $shifts = $template->shifts()
```

Comment:
What if template is null?

Reply:
Or is it guaranteed by migration?

Reply:
this is guarded by migration
i'm planning to extract a templatemanagement, so schedules and templates are diffrent kind of objects